lib/codit/population/person.py

```diff
@@ -40,14 +39,13 @@ def attack(self, other, days):
             self.infectious_attack(other, days)
 
     def infectious_attack(self, other, days):
-        if not other.infected:
+        if self.disease not in other.immunities:
             if random.random() < self.disease.pr_transmit_per_day * days:
                 other.set_infected(self.disease, infector=self)
                 self.victims.add(other)
 
     def set_infected(self, disease, infector=None):
-        assert self.disease is None
-        self.infected = True
+        self.immunities.add(disease)
         self.infectious = True
```

Comment:
I think the self.infected should be kept, as immunities could come from vaccines or disease. If introduces vaccines, infected and vaccined should be separate status with similarities and differences. E.g. Infection would only add one disease strain into self.immunities, while vaccine can add more than one strains into self. immunities, and vaccined person is not infectious.

Reply:
```python
@property
def self.infected():
    return len(self.covid_experiences) > 0
```